Detect OS fails due to mismatching release name on RHEL 8

The release name for RHEL 8 nodes seems to have changed from;

`Red Hat Enterprise Linux Server release 8.9`
to
`Red Hat Enterprise Linux release 8.9`

Hence the token count will raise IndexErrors

```
/tmp/f_scout_ci/actions-runner-01/_temp/test-kenv/root/lib64/python3.8/site-packages/subscript/runrms/runrms.py:291: in __init__
    self.detect_os()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <subscript.runrms.runrms.RunRMS object at 0x7f4bdfc0aac0>

    def detect_os(self):
        """Detect operating system string in runtime, just use default if not found."""
        if RHEL_ID.is_file():
            with open(RHEL_ID, "r", encoding="utf-8") as buffer:
>               major = buffer.read().split(" ")[6].split(".")[0].replace("'", "")
E               IndexError: list index out of range
```

--- a/src/subscript/runrms/runrms.py
+++ b/src/subscript/runrms/runrms.py
@@ -308,9 +308,14 @@ class RunRMS:
         """Detect operating system string in runtime, just use default if not found."""
         if RHEL_ID.is_file():
             with open(RHEL_ID, "r", encoding="utf-8") as buffer:
-                major = buffer.read().split(" ")[6].split(".")[0].replace("'", "")
-                self.osver = "x86_64_RH_" + str(major)
-                logger.debug("RHEL version found in %s", RHEL_ID)
+                tokens = buffer.read().split()
+                for t in tokens:
+                    if "." in t:
+                        major = t.split(".")[0]
+                        self.osver = f"x86_64_RH_{major}"
+                        logger.debug("RHEL version found in %s", RHEL_ID)
+                        return
+            raise ValueError("Could not detect RHEL version")
 
     def do_parse_args(self, args):
         """Parse command line args."""

--- a/tests/test_runrms.py
+++ b/tests/test_runrms.py
@@ -128,7 +128,8 @@ def test_runlogger(tmp_path):
         ("Red Hat Enterprise Linux Server release 6.0 (bar)", "x86_64_RH_6"),
         ("Red Hat Enterprise Linux Server release 7.2 (Maipo)", "x86_64_RH_7"),
         ("Red Hat Enterprise Linux Server release 8.2 (foo)", "x86_64_RH_8"),
-        pytest.param("foobar", None, marks=pytest.mark.xfail(raises=IndexError)),
+        ("Red Hat Enterprise Linux release 8.9 (Ootpa)", "x86_64_RH_8"),
+        pytest.param("foobar", None, marks=pytest.mark.xfail(raises=ValueError)),
     ],
 )
 def test_detect_os(os_id, expected, tmp_path, mocker):

